packages/py-moose-lib/moose_lib/internal.py

```
@@ -830,6 +857,86 @@ def _convert_engine_instance_to_config_dict(engine: "EngineConfig") -> EngineCon
+    from moose_lib.dmv2.olap_table import OlapTable
+    from moose_lib.dmv2.view import View
+
+    if config.source_table is not None:
+        src = config.source_table
+        database = None
+        if isinstance(src, OlapTable):
+            database = src.config.database
+        elif isinstance(src, View):
+            database = getattr(src, "database", None)
+        return {"type": "TABLE", "table": src.name, "database": database}
+    elif config.source_query is not None:
```

Ruff (0.15.7)

[warning] Unnecessary `elif` after `return` statement

Remove unnecessary `elif`

(RET505)